pkg/build/builder/common.go

```diff
@@ -84,15 +98,9 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 			},
 		},
 	}
-
+	// NOTE: not sure that this is needed anymore
 	// Reset ResourceVersion to avoid a conflict with other updates to the build
 	build.ResourceVersion = ""
```

Comment:
@Miciah +1, either we should always clear this because we never want to worry about conflicts, or we should never be updating it because we should always care about conflicts.

@csrwng want to shed some light on the history here?